drivers/google/create.go

```diff
@@ -49,6 +50,19 @@ func (p *provider) Create(ctx context.Context, opts autoscaler.InstanceCreateOpt
 		}
 	}
 
+	if p.region != "" {
+		// If region was specified, use region to choose a random zone
+		region, err := p.service.Regions.Get(p.project, p.region).Do()
+		if err != nil {
+			logger.WithError(err).
+				Errorln("cannot get region details")
+			return nil, err
+		}
+		randomZoneIndex := rand.Intn(len(region.Zones))
+		zoneSelfLink := region.Zones[randomZoneIndex]
+		p.zone = zoneSelfLink[strings.LastIndex(zoneSelfLink, "/")+1:]
```

Comment:
you could probably simplify as `p.zone = path.Base(zoneSelfLink)`

Reply:
sorry, one gotcha here is that `p.zone` is not going to be thread safe to write to.  do we need to overwrite `p.zone` or can this be stored in a local variable?

Reply:
You are right on both accounts - thanks :) This also uncovered that the operation wait function relied on having `p.zone` and succeeded during tests because I was obviously running that one test synchronously 😅 So I've changed that function to take a `zone` as an argument.